platforms/software/maven/src/main/java/org/gradle/api/publish/maven/MavenArtifact.java

```diff
@@ -15,39 +15,27 @@
  */
 package org.gradle.api.publish.maven;
 
+import org.gradle.api.provider.Property;
 import org.gradle.api.publish.PublicationArtifact;
-import org.gradle.internal.instrumentation.api.annotations.ToBeReplacedByLazyProperty;
-
-import javax.annotation.Nullable;
+import org.gradle.api.tasks.Optional;
+import org.gradle.internal.instrumentation.api.annotations.ReplacesEagerProperty;
 
 /**
  * An artifact published as part of a {@link MavenPublication}.
  */
 public interface MavenArtifact extends PublicationArtifact {
```

Comment:
Looks like we didn't upgrade `PublicationArtifact`. Is there a reason we didn't?